plugins/ConvertStacked/ConvertStacked.cpp

```diff
@@ -134,13 +134,41 @@ class ConvertFromStacked : public GenericVideoFilter
 {
 public:
 
-    ConvertFromStacked::ConvertFromStacked(PClip src, IScriptEnvironment* env) : GenericVideoFilter(src)
+    ConvertFromStacked::ConvertFromStacked(PClip src, int bits, IScriptEnvironment* env) : GenericVideoFilter(src)
     {
-        if (vi.IsYV12()) vi.pixel_type = VideoInfo::CS_YUV420P16;
-        else if (vi.IsYV16()) vi.pixel_type = VideoInfo::CS_YUV422P16;
-        else if (vi.IsYV24()) vi.pixel_type = VideoInfo::CS_YUV444P16;
-        else if (vi.IsY8()) vi.pixel_type = VideoInfo::CS_Y16;
-        else env->ThrowError("ConvertFromStacked: Input stacked clip must be YV12, YV16, YV24 or Y8");
+        if (bits == 10 && vi.IsYV12())
+            vi.pixel_type = VideoInfo::CS_YUV420P10;
+        else if (bits == 10 && vi.IsYV16())
+            vi.pixel_type = VideoInfo::CS_YUV422P10;
+        else if (bits == 10 && vi.IsYV24())
+            vi.pixel_type = VideoInfo::CS_YUV444P10;
+        else if (bits == 10 && vi.IsY8())
+            vi.pixel_type = VideoInfo::CS_Y10;
+        else if (bits == 12 && vi.IsYV12())
+            vi.pixel_type = VideoInfo::CS_YUV420P12;
+        else if (bits == 12 && vi.IsYV16())
+            vi.pixel_type = VideoInfo::CS_YUV422P12;
+        else if (bits == 12 && vi.IsYV24())
+            vi.pixel_type = VideoInfo::CS_YUV444P12;
+        else if (bits == 12 && vi.IsY8())
+            vi.pixel_type = VideoInfo::CS_Y12;
+        else if (bits == 14 && vi.IsYV12())
+            vi.pixel_type = VideoInfo::CS_YUV420P14;
+        else if (bits == 14 && vi.IsYV16())
+            vi.pixel_type = VideoInfo::CS_YUV422P14;
+        else if (bits == 14 && vi.IsYV24())
+            vi.pixel_type = VideoInfo::CS_YUV444P14;
+        else if (bits == 14 && vi.IsY8())
+            vi.pixel_type = VideoInfo::CS_Y14;
+        else if (bits == 16 && vi.IsYV12())
+            vi.pixel_type = VideoInfo::CS_YUV420P16;
+        else if (bits == 16 && vi.IsYV16())
+            vi.pixel_type = VideoInfo::CS_YUV422P16;
+        else if (bits == 16 && vi.IsYV24())
+            vi.pixel_type = VideoInfo::CS_YUV444P16;
+        else if (bits == 16 && vi.IsY8())
+            vi.pixel_type = VideoInfo::CS_Y16;
+        else env->ThrowError("ConvertStackedToNative: Input stacked clip must be YV12, YV16, YV24 or Y8");
```

Comment:
On 1.), I don't know why I didn't think of using avs_num_components to determine whether 3 or 4 planes were needed.

Point 2.) requires some background. The bit allocations you see under the 2.6 part are wonky as it is.  The real story is that the AviSynth demuxer in libavformat was initially rewritten by (one of?) AvxSynth's developers, and since AvxSynth was essentially just AviSynth 2.5 with some of 2.6's API backported to it, the plain avs_bits_per_pixel use meant that it worked for YV12 because it also worked for AviSynth 2.5, and 2.6 hadn't changed anything in that regard.  The rest of the formats being parsed were all packed (YUY2, RGB24, RGB32), so bits_per_pixel returned the correct values.  When I was helping bug test the rewritten demuxer to get it committed, I discovered that you could get 2.6's newer planar formats (YV24, YV16, YV411, and Y8) to work if the actual bpp value was explicitly defined, because avs_bits_per_pixel didn't work with them.

And these explicitly-defined values worked for us, too, even when the high bit depths were added (earlier versions of the FFmpeg patch could open AVS_CS_YUV444P16 and recognize it as yuv444p16 without messing with the bit allocations at all).  It might still work against the current upstream MT, but not against this pull request.  Whatever change occurred to vi.BitsPerPixel in AviSynth+ (or this pull request specifically), YV12 stopped working with avs_bits_per_pixel, causing FFmpeg to crash, but with it explicitly defined to a bit value like the rest of the planar formats, it started working again.  Whatever the change was, it also made the high bit depths stop working with avs_bits_per_pixel.  As above, though, the packed ones were unaffected.  RGB24/32, RGB48/64, and YUY2 were detected correctly.

Basically I just said "screw it" and used avs_bits_per_component since it's dumb math and libavformat doesn't complain about packets being too big for a sample.  I'm not opposed to making it fit the pix_fmts more exactly, but since FFmpeg still supports 2.6, it has to be shielded from the avsplus-exclusive functions, so it'll still be branched like it currently is.  It's probable that `avs_bits_per_pixel * avs_bits_per_component / 8` would have been suggested if I'd submitted the current form of the patch to their mailing list anyway, so it doesn't hurt anything to correct it now.

With 3.), I just hadn't gotten around to adjusting that.  Packed RGB has to be flipped, and it simply doesn't check for planar RGB yet.  It does need flipping too.